[ML] Update http access modules for ECS

sophiec20 opened the pull request:
## Summary

Updates the fields used in the NGINX and Apache web access data recognizer modules to the Elastic Common Schema (ECS) field names.

The ECS modules also differ from the non-ECS versions by replacing the two drilldown dashboards (URL Explorer and Count Explorer) with a single HTTP Access Explorer dashboard for drilldown.

#### Summary of changes from non ECS versions:

- Renamed following fields:
  - event.module:apache
  - event.dataset:access
  - source.address
  - url.original
  - http.response.status_code
  - source.geo.location

- Only use one set of kibana saved objects for all http web access logs
- Rename files from apache2 to apache
- Combined URL explorer into Count explorer dashboard as there was a lot of duplication
- Rename custom urls to Investigate Source IP and Status Code
- Added chart to show overall event rate split by event.module - can tell if multiple datasets are included
- Increased limit for top source ips from 5 to 50
- Add `created_by` field for jobs
- Rename jobs and saved objects to include ecs tag


### Checklist

- [x] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)

### For maintainers

- [x] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

sophiec20 committed:
[ML] Initial commit for apache ecs module

sophiec20 committed:
[ML] Update apache2 module for ECS

Rename following fields
event.module:apache
event.dataset:access
source.address
url.original
http.response.status_code
source.geo.location

Rationalise to only use one set of kibana saved objects for all http web access logs
Rename files from apache
Combined URL explorer into Count explorer dashboard as there was a lot of duplication
Add filter to custom url
Rename custom urls to Investigate Source IP and Status Code
Add chart to show overall event rate split by event.module - can tell if multiple datasets are included
Increase limit for top source ips from 5 to 50
Add created_by to custom setting for telemetry
Rename jobs and saved objects to include ecs tag

Tested side by side against v6 jobs

elasticmachine commented:
Pinging @elastic/ml-ui

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9133/)

sophiec20 committed:
[ML] Rename apache files from hyphen to underscores

sophiec20 committed:
[ML] Further apache renames

Also change custom URLs to lower case to match "View series"
Change created_by to ml-module-apache-access

sophiec20 committed:
[ML] Initial commit of nginx ml module

sophiec20 committed:
[ML] Rename dashboard to generic explorer

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9472/)

sophiec20 committed:
[ML] Further refinement for apache

Rename http_status_code to status_code_rate
Update custom url to use filters instead of lucene query bar

sophiec20 committed:
[ML] Convert apache module to nginx

Copy files, keeping nginx logo
Multiple renames to nginx

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9727/)

sophiec20 committed:
[ML] Make chart legend visible by default

sophiec20 commented:
This pr now contains nginx_ecs and apache_ecs modules and is ready for review. Data set is available for apache, nginx will be ready soon pending new build after https://github.com/elastic/beats/pull/10418.

This pr does not yet remove the non-ecs modules. This will be a separate pr as these non-ecs modules may be useful for side-by-side comparisons.

jgowdyelastic reviewed (approved):
LGTM

peteharverson reviewed (approved):
LGTM. May tweak some of the labels, but I will do this in a follow up PR

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/10038/)
